ci/do_ci.sh

```diff
@@ -232,6 +233,9 @@ elif [[ "$CI_TARGET" == "bazel.compile_time_options" ]]; then
   echo "Building and testing ${TEST_TARGETS}"
   bazel test ${BAZEL_BUILD_OPTIONS} ${COMPILE_TIME_OPTIONS} -c dbg ${TEST_TARGETS} --test_tag_filters=-nofips --build_tests_only
 
+  # Legacy codecs "--define legacy_codecs_in_test=true" should be tested in integration tests with asan.
+  bazel test ${BAZEL_BUILD_OPTIONS} ${COMPILE_TIME_OPTIONS} -c dbg @envoy//test/integration/... --config=clang-asan --build_tests_only
```

Comment:
Can you quantify how much additional time this adds to CI? I'm guessing that due to RBE and caching magic it's not too bad, but worth verifying. 

Other than that concern, I think your approach is a good balance of test coverage and developer overhead. Let's go with it while we sunset the legacy codecs.

Reply:
Sounds good, launched compile_time_options, I'll give you some number increase for my local docker runs of compile_time_options. I hit 2 failures in integration tests though, and just need to figure out if those are flakes or something I did with modifying runtime.

Reply:
Aside do we have a policy on how long something like this sticks around?
I'm inclined to say that since no functional change is intended, we can get rid of the old version faster than we can with the usual runtime guards which involve actual L7 changes.  Thoughts?

Reply:
To Harvey's question about CI, compile time took 38 min, and a nrecent PRs took 35, so probably not that much

Reply:
SGTM